File: sort/insert_sort.py
def insertSort(nums: list):
    n = len(nums)
    # 从第二个元素开始（索引1），因为第一个元素可以看作已排序
    for i in range(1, n):
        current = nums[i]  # 当前要插入的元素
        j = i - 1  # 已排序部分的最后一个位置
        
        # 在已排序部分中从右到左找插入位置
        while j >= 0 and nums[j] > current:
            nums[j + 1] = nums[j]  # 向右移动元素
            j -= 1
        
        # 插入当前元素到正确位置
        nums[j + 1] = current
        print(f"bound {i}: {nums}")

if __name__ == "__main__":
    nums = [10, 3, 1, 8, 4, 7, 6]
    print(f"Before Sorted: {nums}")
    insertSort(nums)
    print(f"After Sorted: {nums}")

File: sort/insert_sort.cpp
#include <iostream>
#include <vector>

using namespace std;
void insertSort(vector<int>& nums) {
    for (int i = 1; i < nums.size(); i++) {
        int cur = nums[i];
        int j = i - 1;
        while (j >= 0 && nums[j] > cur) {
            nums[j + 1] = nums[j];
            j--;
        }
        nums[j + 1] = cur;
        cout << "bound " << i << ": ";
        for (int& num : nums) {
            cout << num << ", ";
        }
        cout << endl;        
    }
}

int main(int argc, char const *argv[])
{
    vector<int> nums{10, 3, 1, 8, 4, 7, 6};
    cout << "Before Sorted: [";
    for (int& num : nums) {
        cout << num << ", ";
    }
    cout << "]" << endl;
    insertSort(nums);
    cout << "After Sorted: [";
    for (int& num : nums) {
        cout << num << ", ";
    }
    cout << "]" << endl;
    return 0;
}
